nilearn/regions/signal_extraction.py

```diff
@@ -154,9 +160,8 @@ def _get_labels_data(
 
     labels_data = _safe_get_data(labels_img, ensure_finite=True)
 
-    labels = list(np.unique(labels_data))
-    if background_label in labels:
-        labels.remove(background_label)
+    if keep_masked_labels:
+        labels = list(np.unique(labels_data))
```

Comment:
This is where we can add a deprecation warning signaling that this behavior is deprecated and that the user should set this parameter to False